server/src/main/java/org/opensearch/index/mapper/TextFieldMapper.java

```diff
@@ -583,7 +583,9 @@ public Query prefixQuery(String value, MultiTermQuery.RewriteMethod method, bool
             }
             Automaton automaton = Operations.concatenate(automata);
             AutomatonQuery query = new AutomatonQuery(new Term(name(), value + "*"), automaton);
-            query.setRewriteMethod(method);
+            if (method != null) {
+                query.setRewriteMethod(method);
```

Comment:
`query.setRewriteMethod` is [removed in Lucene 9.2](https://github.com/apache/lucene/pull/726). I've [opened an issue](https://github.com/opensearch-project/OpenSearch/issues/2896) to remove all references to the method and use [`CONSTANT_SCORE_REWRITE`](https://github.com/apache/lucene/blob/e6fb74f9090db2bc274af94c17d80739697bdc01/lucene/core/src/java/org/apache/lucene/search/MultiTermQuery.java#L81) in places where `null` is passed in as the `method` parameter.